ios-base/Networking/APIClient.swift

```diff
@@ -149,17 +149,20 @@ class APIClient {
   }
   
   //Handle rails-API-base errors if any
-  class func handleCustomError(_ code: Int?, dictionary: [String: Any]) -> NSError? {
-    if let messageDict = dictionary["errors"] as? [String: [String]] {
-      let errorsList = messageDict[messageDict.keys.first!]!
-      return NSError(domain: "\(messageDict.keys.first!) \(errorsList.first!)", code: code ?? 500, userInfo: nil)
-    } else if let error = dictionary["error"] as? String {
-      return NSError(domain: error, code: code ?? 500, userInfo: nil)
-    } else if let errors = dictionary["errors"] as? [String: Any] {
-      let errorDesc = errors[errors.keys.first!]!
-      return NSError(domain: "\(errors.keys.first!) " + "\(errorDesc)", code: code ?? 500, userInfo: nil)
-    } else if dictionary["errors"] != nil || dictionary["error"] != nil {
-      return NSError(domain: "Something went wrong. Try again later.", code: code ?? 500, userInfo: nil)
+  class func findError(inResponse response: [String: Any], code: Int?) -> NSError? {
```

Comment:
why not returning an array of NSErrors and not just the first found ? clients can later choose to show the first one or manage every found error

Reply:
`Alamofire.Response.Result` allows only one error in case of .failure. We could reach an agreement and implement something where we pass the list of errors under some key in the `userInfo` or the list of errors formatted into a single string.

Reply:
Yes, this is something that we have had to manage in many projects. We can open an issue with this to tackle later.